app/controllers/users_controller.rb

```diff
@@ -1,17 +1,18 @@
 class UsersController < ApplicationController
   before_action :authorize_request, except: [:create]
 
-  def index
-    @users = User.all
-    
-    render json: @users.as_json(except: %i[created_at updated_at]), status: :ok
-  end
-
   def show
-    @user = User.where(id: params[:id]).first
+    @user = User.includes(:lists).where(id: params[:id]).first
 
     if @user
-      render json: @user.as_json(except: %i[created_at updated_at]), status: :ok
+      if @user.id == @current_user.id
```

Comment:
Same here, you don't need this logic. See https://scotch.io/tutorials/build-a-restful-json-api-with-rails-5-part-two as an example. Same for other endpoints 😄